lms/djangoapps/courseware/tabs.py

```diff
@@ -39,12 +39,21 @@ class CoursewareTab(EnrolledTab):
     supports_preview_menu = True
 
     @property
-    def link_func(self):
+    # DO NOT APPROVE CODE REVIEW WITH THIS DEFAULT!!!
+    # TODO: LEARNER-1604: Fix this by removing fake default
+    def link_func(self, course_id=None):
         """
         Returns a function that computes the URL for this tab.
+
+        Arguments:
+            course_id (CourseKey): The course id of the current course.
         """
-        request = RequestCache.get_current_request()
-        url_name = default_course_url_name(request)
+
+        # TODO: LEARNER-1604: Don't Approve until this has been removed
```

Comment:
Reviewer: This is what @dianakhuang is working on.  Hopefully all tests will pass once this is working correctly.